### PR TITLE
Update from the React repo to use new protocol

### PR DESCRIPTION
**Pull request by sebmarkbage**

### CODE REVIEW

File: src/ReactARTFiber.js [hunk 1]
```diff
@@ -18,8 +18,8 @@ require('art/modes/current').setCurrent(
 const Mode = require('art/modes/current');
 const Transform = require('art/core/transform');
 const invariant = require('fbjs/lib/invariant');
-const React = require('react');
-const ReactFiberReconciler = require('react-dom/lib/ReactFiberReconciler');
+const React = require('React');
+const ReactFiberReconciler = require('ReactFiberReconciler');
```
Comment (bvaughn) [inline src/ReactARTFiber.js]: Looks like these lines made it in accidentally?

File: src/ReactARTFiber.js [hunk 2]
```diff
@@ -489,6 +485,10 @@ const ARTRenderer = ReactFiberReconciler({
     // Noop
   },
 
+  getChildHostContext() {
+    return null;
+  },
+
```
Comment (bvaughn) [inline src/ReactARTFiber.js]: Why the arbitrary relocation? Methods were alpha-sorted before (for easier readability).

File: src/ReactARTFiber.js (hunk 1 above)
Comment (sebmarkbage) [inline src/ReactARTFiber.js]: Fixed in a follow up.

File: src/ReactARTFiber.js (hunk 2 above)
Comment (sebmarkbage) [inline src/ReactARTFiber.js]: Copied from the react repo. Now they're the same. :)

Comment (bvaughn) [inline src/ReactARTFiber.js]: Ah, I see. NBD. :grin: Alpha-sorting is just nice to have when visually comparing changes across renderers.

**Pull request merged**